File: C++/BASIC MATHS REVISION/armstrong_number.cpp
#include<iostream>
using namespace std;    
int main(){
    int n;
    int digit=n;
    int sum = 0;
    int revNum = 0;
    cin>>n;
    while(n>0){
        int digit = n % 10; // Get the last digit
        sum += digit * digit * digit; // Calculate the sum of cubes
        n= n / 10; // Remove the last digit
    }
    cout << "Reversed Number: " << revNum << endl;
    if(digit == sum){
        cout << "The number is a armstrong number." << endl;
    } else {
        cout << "The number is not a armstrong number." << endl;
    }
    return 0;
}

File: JAVA/BASIC MATHS REVISION/armstrong_number.java
import java.util.Scanner;

public class armstrong_number {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int originalNum = n;
        int sum = 0;
        int revNum = 0;
        
        while (n > 0) {
            int digit = n % 10;
            sum += digit * digit * digit;
            revNum = revNum * 10 + digit;
            n = n / 10;
        }
        
        System.out.println("Reversed Number: " + revNum);
        if (originalNum == sum) {
            System.out.println("The number is an Armstrong number.");
        } else {
            System.out.println("The number is not an Armstrong number.");
        }
    }
}
